### ISSUE TITLE
odf: R cannot convert 1.0D1 style numbers to numeric

### ISSUE THREAD

**Issue by dankelley** [dankelley/oce#1158]
I ran into this in looking at 1154. With that test file, I get a warning about coercion to numeric. Converting the warning to an error and then using `traceback`, I see that the problem is in interpreting a missing value using `as.numeric(findInHeader("NULL_VALUE", lines))` because the `lines` value is
```
   NULL_VALUE=-.99000000D+02,
```
Oh, my, this test file is odd for so many reasons. (1: no conductivity or salinity; 2: every line starts with a space; 3: numerical values are given in "D" format, which I am quit sure I've not seen in the many hundreds of ODF files I've worked with)

A solution would be to cause `findInHeader` to `gsub("D\\+([0-9]{2})", "e+\\1", line)` before processing further.

**Comment by richardsc** [comment]
Ah, yes. I also saw those warnings (on a suite of recently downloaded, archived ODF files), but was unable to trace where it was coming from (admittedly, I didn't try very hard).

Now that you point that out, I did also notice that the data were full of missing values (e.g. -99's). I was going to open an issue on that but instead just tossed in:
```r
        for (field in names(d@data)) {
            if (!(field %in% 'theta')) {
                bad <- d[[field]] == -99
                d[[field]][bad] <- NA
            }
        }
```
which coincidentally was what led to discovering Issue #1154

**Comment by dankelley** [comment]
Fixed in `develop` commit af087784fd33a045dc37f3bd7883ca146ecdb58d

**Comment by dankelley** [comment]
`read.odf()` is actually handling these missing values incorrectly. The ODF spec seems to permit using a different code for each variable, but `read.odf()` is assuming that the first lines matching to `NULL_VALUE` is giving a missing-value code for the *entire* file. I don't really want to spend the hour it might take to write code for individual missing values for different variables, however, since I don't think I've ever seen a file configured with different values. (This is a case of leaving a ragged thread in the cloth because it seems more important to fix a gaping tear somewhere else.)
